### PR TITLE
Implement `cargo clippy --all`

### PR DESCRIPTION
**Pull request by bjgill**
This implements workspace support for `cargo clippy` by running clippy
over all packages in the workspace (in serial).

This should probably be parallelised in future (as `cargo build --all`).

See #1707

### CODE REVIEW

**Comment by bjgill** [comment]
And this fails to compile due to #1970...

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ script:
  - cargo test --features debugging
  - mkdir -p ~/rust/cargo/bin
  - cp target/debug/cargo-clippy ~/rust/cargo/bin/cargo-clippy
- - PATH=$PATH:~/rust/cargo/bin cargo clippy -- -D clippy
+ - PATH=$PATH:~/rust/cargo/bin cargo clippy --all -- -D clippy
```
Comment (oli-obk) [inline .travis.yml]: I think this means we can remove the line below?

Comment (bjgill) [inline .travis.yml]: Very true - removed

**Comment by oli-obk** [comment]
Thanks!

**Pull request merged**